packages/web/src/components/apps/use-package-json.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { createContext, useCallback, useContext, useState } from 'react';
+import { OutputType } from '@srcbook/components/src/types';
+import { AppChannel } from '@/clients/websocket';
+import {
+  DependenciesInstallLogPayloadType,
+  DependenciesInstallStatusPayloadType,
+} from '@srcbook/shared';
+import { useLogs } from './use-logs';
+
+type NpmInstallStatus = 'idle' | 'installing' | 'complete' | 'failed';
+
+export interface PackageJsonContextValue {
+  npmInstall: (packages?: string[]) => void;
+  status: NpmInstallStatus;
+  installing: boolean;
+  failed: boolean;
+  output: Array<OutputType>;
+}
```

Comment:
A lot of this interface should look familiar to the similar context that exists for srcbook notebooks - I originally tried to go down the path of making that reusable, but that proved to be too challenging to do quickly.

Reply:
Sounds good. We can take both later and extract an optimal interface, but this is not urgent at all. Good call